client/ayon_flame/api/plugin.py

```diff
@@ -631,46 +637,156 @@ def get_native_colorspace(cls, input_colorspace):
 
-class OpenClipSolver(flib.MediaInfoFile):
-    create_new_clip = False
+    def load(self, context, name, namespace, options):
+        # get flame objects
+        fproject = flame.project.current_project
+        self.fpd = fproject.current_workspace.desktop
 
-    log = log
+        # load clip to timeline and get main variables
+        version_entity = context["version"]
+        representation_name = context["representation"]["name"]
+        project_name = context["project"]["name"]
 
-    def __init__(self, openclip_file_path, feed_data, logger=None):
-        self.out_file = openclip_file_path
+        if not context["representation"]["context"].get("output"):
+            self.clip_name_template = self.clip_name_template.replace(
+                "output", "representation"
+            )
+        clip_name = StringTemplate(self.clip_name_template).format(
+            self._get_formatting_data(context, options)
+        )
 
-        # replace log if any
-        if logger:
-            self.log = logger
+        # create workfile path
+        workfile_dir = os.environ["AYON_WORKDIR"]
+        openclip_dir = os.path.join(
+            workfile_dir, clip_name
+        )
+        openclip_path = os.path.join(
+            openclip_dir, clip_name + ".clip"
+        )
+        if not os.path.exists(openclip_dir):
+            os.makedirs(openclip_dir)
+
+        clip_solver = OpenClipSolver(
+            openclip_path, self.layer_rename_patterns
+        )
+        rename_clip = clip_solver.out_clip_data is None
+        message_args = None
+        if not version_entity["taskId"]:
+            message_args = (
+                "Import Warning",
+                "It looks like the task entity for this version has been "
+                "removed from the database.\n"
+                "Unable to import any other versions alongside the current "
+                "selection.\nThis may result in unordered versions within the"
+                " clip. Deleting the clip and reloading it will resolve this "
+                "issue.",
+                "warning",
+                ["OK"],
+            )
+            all_versions = [version_entity]
+        else:
+            all_versions = ayon_api.get_versions(
+                project_name,
+                product_ids=[version_entity["productId"]],
+                task_ids=[version_entity["taskId"]],
+            )
 
-        # new feed variables:
-        feed_path = feed_data.pop("path")
+        for version in all_versions:
+            representation = ayon_api.get_representation_by_name(
+                project_name,
+                representation_name,
+                version_id=version["id"],
+            )
+
+            version_context = deepcopy(context)
+            version_context["version"] = version
+            version_context["representation"] = representation
+
+            version_name = f"v{version['version']:03}"
```

Comment:
Also handles hero version.
```suggestion
            version_name = version["name"]
```